### PR TITLE
Make sequence number order correspond to field number order

### PR DESCRIPTION
**Pull request by kwahlin**
So that we get
240-01
245-02
264-03
700-04
700-05

instead of for example
240-04
245-03
264-05
700-01
700-02

A nice to have that we can consider bringing in as a hotfix.

- [x] integTest

### CODE REVIEW

File: whelk-core/src/main/groovy/whelk/converter/marc/RomanizationStep.groovy [hunk 1]
```diff
@@ -144,65 +144,60 @@ class RomanizationStep extends MarcFramePostProcStepBase {
...
+                            [
+                                    (TYPE)          : 'marc:Bib880',
+                                    (PART_LIST)     : [[(FIELDREF): fieldNumber]] + field[SUBFIELDS],
+                                    (BIB880 + '-i1'): field[IND1],
+                                    (BIB880 + '-i2'): field[IND2]
+                            ]
 
-                hasBib880.add(bib880)
-                fieldRefs.add(ref)
+                    bib880ToRef[bib880] = ref
```
[the remark this answers is not in the document]
Comment (kwahlin) [reply whelk-core/src/main/groovy/whelk/converter/marc/RomanizationStep.groovy:186]: We should never get exact duplicates of `bib880`'s though? The subfield values should always differ.

[the remark this answers is not in the document]
Comment (olovy) [reply whelk-core/src/main/groovy/whelk/converter/marc/RomanizationStep.groovy:186]: The only thing that is varying is `fieldNumber`:
```
[
                        occurenceNumber: hasBib880.size() + 1,
                                    (TYPE)          : 'marc:Bib880',
                        path: path.dropRight(1)
                                    (PART_LIST)     : [[(FIELDREF): fieldNumber]] + field[SUBFIELDS],
                )
                                    (BIB880 + '-i1'): field[IND1],

                                    (BIB880 + '-i2'): field[IND2]
                def scriptCode = marcScript(tLang)
                            ]
```

[the remark this answers is not in the document]
Comment (kwahlin) [reply whelk-core/src/main/groovy/whelk/converter/marc/RomanizationStep.groovy]: Fixed: [35a8bea](https://github.com/libris/librisxl/pull/1238/commits/35a8bea1240883b17d232cc7af7109cb83a4b618)

**Review by olovy — APPROVED** [review]
LGTM!

**Pull request merged**